src/components/PopupWithForm.js

```diff
@@ -16,16 +17,23 @@ export default class PopupWithForm extends Popup {
     return inputValues;
   }
 
-  sendInputValues() {
-    return this._getInputValues();
+  _renderSaving(isSaving) {
+    const submitButton = this._popupForm.querySelector(".modal__save");
+    if (isSaving) {
+      submitButton.textContent = "Saving...";
```

Comment:
**Could be improved**

It’s better to make method `renderLoading` inside class `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```

Reply:
Okay I changed it to called renderLoading instead of renderSaving, thanks for name suggestion. One of the things I struggle with is finding a good name for some stuff.

Reply:
The main idea for my advice is to get the initial button text like that

```js

    // fix the initial button text only once in the constructor
    this._defaultSubmitButtonText = this._submitBtn.textContent
```


So, you don't even need to pass it manually into the constructor